massa-protocol-worker/src/protocol_worker.rs

```diff
@@ -929,7 +937,10 @@ impl ProtocolWorker {
             ops.store_operations(new_operations.into_values().collect());
 
             // Add to pool, propagate when received outside of a header.
-            self.pool_controller.add_operations(ops);
+            self.pool_controller.add_operations(ops.clone());
+
+            // Request propagation
+            self.propagate_ops(ops).await;
```

Comment:
I think we need to add a `propagate` boolean, to `note_operations_from_node`, and set it to `true` when calling it from `on_block_full_operations_received`, and `false` otherwise.

Reply:
Since separate op propagation, we don't need to do this anymore. We always propagate ops to all the nodes that don't know them (whether they come from blocks or naked). And if a node is missing some ops to fill up a block, they can ask them explicitly.